src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/net/ReceivedFlowUnitStore.java

```diff
@@ -100,9 +100,12 @@ public boolean enqueue(final String graphNode, final FlowUnitMessage flowUnitMes
   /**
    * Drains out all the flow units for all nodes.
    */
-  public void drainAll() {
+  public ImmutableList<FlowUnitMessage> drainAll() {
+    List<FlowUnitMessage> drained = new ArrayList<>();
     for (final String graphNode : flowUnitMap.keySet()) {
-      drainNode(graphNode);
+      ImmutableList<FlowUnitMessage> messages = drainNode(graphNode);
+      drained.addAll(messages);
     }
+    return ImmutableList.copyOf(drained);
```

Comment:
we can probably avoid this copy here, right ? Although the test is the only one that needs it, the release code also pays the price for it.

Reply:
Done